### ISSUE TITLE
[MASSEMBLY-709] When assembling a zip on windows duplicate files are added to the assembly

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#923]
**[Jason Lemay](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jasonl)** opened **[MASSEMBLY-709](https://issues.apache.org/jira/browse/MASSEMBLY-709?redirect=false)** and commented

When assembling a zip where duplicate files are copied to the output directory the default behavior is for the first file to copy and the remaining ones to be skipped. When building a project on windows this is not the behavior. On Windows all the duplicate files are added to the final zip assembly.

This was tested on OSX, CentOS, and Windows Server 2009 R2.

Using OSX with these settings:

```
Apache Maven 3.0.5 (r01de14724cdef164cd33c7c8c2fe155faf9602da; 2013-02-19 05:51:28-0800)
Maven home: /usr/local/Cellar/maven30/3.0.5/libexec
Java version: 1.7.0_60, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_60.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.9.4", arch: "x86_64", family: "mac"
```

Running the mvn build as such:

```
mvn -X clean install
```

You end up with this logged to the console:

```
[INFO] Building zip: /Users/jasonl/Desktop/sample/target/sample.zip
[DEBUG] adding directory sample/
[DEBUG] adding directory sample/conf/
[DEBUG] adding entry sample/conf/ConfFile.txt
[DEBUG] adding entry sample/file.txt
[DEBUG] sample/conf/ConfFile.txt already added, skipping
```

When the final zip is examined there are no duplicate files. The plugin worked as intended. The same results were obtained in the CentOS test.

The problem arises when you build on windows.

Building with these settings:

```
Apache Maven 3.0.5 (r01de14724cdef164cd33c7c8c2fe155faf9602da; 2013-02-19 05:51:28-0800)
Maven home: C:\bin\apache-maven-3.0.5
Java version: 1.7.0_60, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk1.7.0_60\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows server 2008 r2", version: "6.1", arch: "amd64", family: "windows"
```

And running the same mvn command:

```
mvn -X clean install
```

The following is output to the console:

```
[INFO] Building zip: C:\Users\Administrator\Desktop\sample\target\sample.zip
[DEBUG] adding directory sample/
[DEBUG] adding directory sample/conf/
[DEBUG] adding entry sample/conf/ConfFile.txt
[DEBUG] adding entry sample/file.txt
[DEBUG] adding entry sample/conf/ConfFile.txt
```

As you can see the assembly did not skip the second ConfFile.txt addition. When the final zip assembly is examined there is infact 2 ConfFile.txt files under the conf directory.

Attached is the sample I used.

---

**Affects:** 2.4, 2.4.1

**Attachments:**
- [sample.zip](https://issues.apache.org/jira/secure/attachment/12714329/sample.zip) (_2.41 kB_)

**Issue Links:**
- [MASSEMBLY-285](https://issues.apache.org/jira/browse/MASSEMBLY-285) regression: duplicate files added to the assembly

**Comment by jira-importer** [comment]
**[Jason Lemay](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jasonl)** commented

Linked MASSEMBLY-285. These are similar issues, but 285 is supposidly fixed. This maybe a regression.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Is it possible to check if the same behaviour happens if you use maven-assembly-plugin 2.4.1 instead of 2.4 ?

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

I have to acknowledge the wrong behaviour with version 2.4.1 as well.

**Comment by jira-importer** [comment]
**[Jason Lemay](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jasonl)** commented

Sorry about the late reply. Yes I can also confirm it happens in 2.4.1.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Hey no problem.  Thank you for support. So unfortunately we have a real bug.. ;-(. Ok. Let's fix it.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

I have tested the current trunk with the example project and unfortunately it has not changed.

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Fixed in plexus io 2.2
